### PR TITLE
Canonicalise IPv6 address before searching for peers

### PR DESCRIPTION
**Pull request by TheMysteriousX**
JunOS (and probably other platforms) compress the peer identifier, but the code searches the uncompressed database field.

The existing behaviour isn't great either - it clobbers the value of bgpPeerIdentifier.

So, to fix it this change saves an IP object and operates on that, then only uses the pretty value for rendering - SQL queries use the unmodified version.

Without this change, none of our IPv6 peers have their neighbour linked on the BGP page.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Jellyfrog** [comment]
Isn’t the correct fix to make sure to save all data uncompressed ?

**Comment by TheMysteriousX** [comment]
I don't think so - storing what the device returns via SNMP in the database and transforming it when it's accessed seems correct.

File: includes/html/pages/device/routing/bgp.inc.php [hunk 1]
```diff
@@ -121,6 +121,9 @@
     unset($alert);
     unset($peerhost, $peername);
 
+    // load the peer identifier into an object
+    $peer['bgpPeerIdentifierObject'] = IP::parse($peer['bgpPeerIdentifier'], true);
```
Comment (murrant) [inline includes/html/pages/device/routing/bgp.inc.php:125]: Why are we setting this in the $peer array and not as a regular variable?

Comment (TheMysteriousX) [reply includes/html/pages/device/routing/bgp.inc.php:125]: It's a mutation of an existing value of the array, so my preference is to keep it within the array.

Happy to change it to an individual variable.

Comment (murrant) [reply includes/html/pages/device/routing/bgp.inc.php:125]: It's fine, I was worried about making future code changes difficult, but looking at this, it needs to be thrown away and created fresh when ported to Laravel anyway...

**Review by murrant — DISMISSED** [review]

**Review by murrant — APPROVED** [review]

**Pull request merged**